src/System.Security.Cryptography.Xml/tests/EncryptedXmlTest.cs

```diff
@@ -232,13 +428,68 @@ public void GetDecryptionKey_EncryptedDataNull()
         }
 
         [Fact]
+        public void GetDecryptionKey_NoEncryptionMethod()
+        {
+            EncryptedData edata = new EncryptedData();
+            edata.KeyInfo = new KeyInfo();
+            edata.KeyInfo.AddClause(new KeyInfoEncryptedKey(new EncryptedKey()));
+            EncryptedXml exml = new EncryptedXml();
+            Assert.Throws<CryptographicException>(() => exml.GetDecryptionKey(edata, null));
+        }
+
+        [Fact]
         public void GetDecryptionKey_StringNull()
         {
             EncryptedXml ex = new EncryptedXml();
             Assert.Null(ex.GetDecryptionKey(new EncryptedData(), null));
         }
 
         [Fact]
+        public void GetDecryptionKey_KeyInfoName()
+        {
+            using (Aes aes = Aes.Create())
+            {
+                EncryptedData edata = new EncryptedData();
+                edata.KeyInfo = new KeyInfo();
+                edata.KeyInfo.AddClause(new KeyInfoName("aes"));
+
+                EncryptedXml exml = new EncryptedXml();
+                exml.AddKeyNameMapping("aes", aes);
+                SymmetricAlgorithm decryptedAlg = exml.GetDecryptionKey(edata, null);
+
+                Assert.Equal(aes.Key, decryptedAlg.Key);
+            }
+        }
+
+        [Fact]
+        public void GetDecryptionKey_CarriedKeyName()
+        {
+            using (Aes aes = Aes.Create())
+            {
+                EncryptedData edata = new EncryptedData();
+                edata.KeyInfo = new KeyInfo();
+                edata.KeyInfo.AddClause(new KeyInfoName("aes"));
+
+                EncryptedKey ekey = new EncryptedKey();
+                byte[] encKeyBytes = EncryptedXml.EncryptKey(aes.Key, aes);
```

Comment:
For all positive EncryptedKey tests, could we use two different AES instances (possibly with two different sizes or other symmetric algorithm) to avoid errors where we could accidentally encrypt the keys in reverse order?